builder/proxmox/common/config.go

```diff
@@ -606,7 +648,27 @@ func (c *Config) Prepare(upper interface{}, raws ...interface{}) ([]string, []st
 
 	// Default qemu_agent to true
 	if c.Agent != config.TriFalse {
-		c.Agent = config.TriTrue
+		warnings = append(warnings, "qemu_agent is deprecated and will be removed in a future release. define QEMU agent settings in a qemu_guest_agent block instead")
```

Comment:
I would suggest we only print that warning if `qemu_agent` is unset here, not if it's not explicitly `false`, otherwise we'll see it all the time unless we add `qemu_agent = false` to the template.

Reply:
It occurred to me that we don't need to set a default value for `c.Agent` given the replacement `c.GuestAgent.Enabled` has a default set further down, so I've changed the logic to warn and convert the value only if the `c.Agent` is not Unset.